restapi/src/main/java/io/stargate/web/docsapi/service/ReactiveDocumentService.java

```diff
@@ -489,12 +498,238 @@ public ReactiveDocumentService(
+              if (arrayData.size() == 0) {
+                throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_ARRAY_POP_OUT_OF_BOUNDS);
+              }
+              JsonNode value = arrayData.get(arrayData.size() - 1);
+              Object finalValue;
+              if (value.isDouble() || value.canConvertToLong()) {
+                double dblValue = value.asDouble();
+                long longValue = (long) dblValue;
+                if ((double) longValue == value.asDouble()) {
+                  finalValue = longValue;
+                } else {
+                  finalValue = dblValue;
+                }
+              } else if (value.isTextual()) {
+                finalValue = value.asText();
+              } else if (value.isBoolean()) {
+                finalValue = value.asBoolean();
+              } else if (value.isObject()) {
+                finalValue = objectMapper.treeToValue(value, Map.class);
+              } else if (value.isArray()) {
+                finalValue = objectMapper.treeToValue(value, List.class);
+              } else {
+                finalValue = null;
+              }
+              arrayData.remove(arrayData.size() - 1);
+              return Pair.with(arrayData, finalValue);
+            });
+  }
+
+  public Maybe<DocumentResponseWrapper<? extends JsonNode>> executeBuiltInFunction(
+      DocumentDB db,
+      String keyspace,
+      String collection,
+      String id,
+      ExecuteBuiltInFunction funcPayload,
+      List<PathSegment> path,
+      ExecutionContext context)
+      throws UnauthorizedException {
+    return Maybe.defer(
+        () -> {
+          List<String> pathString =
+              path.stream().map(seg -> seg.getPath()).collect(Collectors.toList());
+          Maybe<JsonNode> result = null;
+          BuiltInApiFunction function = BuiltInApiFunction.fromName(funcPayload.getOperation());
+          if (function == BuiltInApiFunction.ARRAY_PUSH) {
+            result =
+                handlePush(
+                    db, keyspace, collection, id, funcPayload.getValue(), pathString, context);
+          } else if (function == BuiltInApiFunction.ARRAY_POP) {
+            result = handlePop(db, keyspace, collection, id, pathString, context);
+          }
+          if (result == null) {
+            throw new IllegalStateException(
+                "Invalid operation found at execution time: " + function.name);
+          }
+          return result.map(
+              json -> new DocumentResponseWrapper<>(id, null, json, context.toProfile()));
+        });
+  }
+
+  private Maybe<JsonNode> handlePush(
+      DocumentDB db,
+      String keyspace,
+      String collection,
+      String id,
+      Object valueToPush,
+      List<String> pathString,
+      ExecutionContext context) {
+    return getArrayAfterPush(db, keyspace, collection, id, pathString, valueToPush, context)
+        .map(
+            jsonArray -> {
+              if (jsonArray == null) {
```

Comment:
I don't think mapper will ever pass a null here?

Reply:
Yea I forgot to remove this, it will never be null